### PR TITLE
added tests

### PR DESCRIPTION
**Pull request by sjackle**

### CODE REVIEW

File: types/pmx/tcr/id.schema.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      "title": "Device Type",
+      "type": "string",
+      "description": "PMX TCR Device Type"
+    },
+    "licence": {
+      "title": "Licence",
+      "type": "string",
+      "description": "Installed Licence Type (Functionality Level)"
+    },
+    "speedclass": {
```
Comment (mariusgiger) [inline types/pmx/tcr/id.schema.json:19]:
```suggestion
    "speedClass": {
```

File: types/pmx/tcr/uplink.js [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+/**
+* Decode uplink
+* @param {Object} input - An object provided by the IoT Flow framework
+* @param {number[]} input.bytes - Array of bytes represented as numbers as it has been sent from the device
+* @param {number} input.fPort - The Port Field on which the uplink has been sent
+* @param {Date} input.recvTime - The uplink message time recorded by the LoRaWAN network server
+* @returns {DecodedUplink} The decoded object
+*/
+function decodeUplink(input) {
```
Comment (mariusgiger) [inline types/pmx/tcr/uplink.js:209]: possibly remove the empty function

File: types/pmx/tcr/uplink.js [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+        topic = "cat4";
+        data.LTRCounter = cnt.cat4_ltr_cnt;
+        data.LTRSpeed = cnt.cat4_ltr_spd;
+        data.RTLCounter = cnt.cat4_rtl_cnt;
+        data.RTLSpeed = cnt.cat4_rtl_spd;
+        time.setHours(cnt.cat4_hours);
+        time.setMinutes(cnt.cat4_minutes);
+        break;
+    }
+    emit("sample", { data, topic });
```
Comment (mariusgiger) [inline types/pmx/tcr/uplink.js:316]: in order to emit a sample at a timestamp, change this to `emit("sample", { data, topic, timestamp });`

File: types/pmx/tcr/cat1.schema.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+  "$id": "https://akenza.io/pmx/tcr/cat1.schema.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "processingType": "uplink_decoder",
+  "topic": "cat1",
+  "title": "Filter Cat 1 Values",
+  "properties": {
+    "LTRCounter": {
```
Comment (mariusgiger) [inline types/pmx/tcr/cat1.schema.json:9]: usually we are using the following convention: camelCase properties should always start with a lower case character even for acronyms

**Review by mariusgiger — COMMENTED** [review]
Hi @sjackle, thanks a lot for the contribution! 

A few minor suggestions from our side otherwise the PR looks good.

Please be sure to also include an entry in the changelog file.

File: types/pmx/tcr/uplink.js (hunk 2 above)
Comment (sjackle) [reply types/pmx/tcr/uplink.js:209]: removed

**Comment by sjackle** [comment]
We implemented your suggestions. Tests rund without errors.

**Review by J2thatsme — APPROVED** [review]
Looks great now with the changes. You should be able to merge it.

**Pull request merged**